### PR TITLE
Superposition of Atomic Potentials

### PR DESCRIPTION
**Pull request by susilehtola**
## Description
This PR implements the Superposition of Atomic Potentials guess that I described last fall in arXiv:1810.11659 and doi:10.1021/acs.jctc.8b01089.

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [x] The SAP guess can be used to start SCF calculations.

## Questions
- [ ] Screening? For yuge systems, one could truncate the calculation of the SAP potential to the practical infinity of the atomic calculation (20 bohr), since the potential is a hard zero outside of it.

## Checklist
- [x] Tests added for any new features
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: samples/scf-guess/input.dat [hunk 1]
```diff
@@ -27,6 +27,9 @@ energy('scf')
 set guess huckel
 energy('scf')
 
+set guess sap
+energy('scf')
```
Comment (loriab) [inline samples/scf-guess/input.dat]: Did you mean to add to the tests in `tests/scf-guess/` instead? This file is autogenerated from tests/ by stripping out the `#TEST` lines.

File: psi4/src/read_options.cc [hunk 2]
```diff
@@ -1248,8 +1248,8 @@ int read_options(const std::string &name, Options &options, bool suppress_printi
         options.add_double("INTS_TOLERANCE", 0.0);
         /*- The type of guess orbitals.  Defaults to ``READ`` for geometry optimizations after the first step, to
           ``CORE`` for single atoms, and to ``SAD`` otherwise. The ``HUCKEL`` guess employs on-the-fly calculations
-          like SAD, as described in doi:10.1021/acs.jctc.8b01089. -*/
-        options.add_str("GUESS", "AUTO", "AUTO CORE GWH SAD HUCKEL READ");
+          like SAD, as described in doi:10.1021/acs.jctc.8b01089 which also rescribes the SAP guess. -*/
```
[the remark this answers is not in the document]
Comment (susilehtola) [reply psi4/src/read_options.cc]: Fixed

File: psi4/src/psi4/libfock/sap.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
+/*
+  Copyright (c) 2019, Susi Lehtola
```
[the remark this answers is not in the document]
Comment (susilehtola) [reply psi4/src/psi4/libfock/sap.h]: Sure

File: samples/scf-guess/input.dat (hunk 1 above)
Comment (susilehtola) [reply samples/scf-guess/input.dat]: I did.

**Comment by susilehtola** [comment]
OK, I was able to find the issues myself even though gdb was of little help.

**Comment by susilehtola** [comment]
Looks like Travis is still feeling ill: the failure is caused by
```
>           raise ResourceError(f"MP2D version '{self.get_version()}' too old. Please update to at least '1.1'.")
E           qcengine.exceptions.ResourceError: MP2D version '' too old. Please update to at least '1.1'.
```

**Comment by susilehtola** [comment]
> Thanks so much! This looks great. I think a little bit of description around line 446 of https://github.com/psi4/psi4/blob/master/doc/sphinxman/source/scf.rst would help users find this option; other than that, LGTM

Done.

**Comment by andysim** [comment]
Thanks for the quick response - the docs are very helpful.  I'll merge this once the CI has finished.

**Pull request merged**